### ISSUE TITLE
[BUG] Spring generator uses old SpringFox & Swagger dependencies

### ISSUE THREAD

**Issue by dkirrane** [OpenAPITools/openapi-generator#4245]
##### Description

I'm using `openapi-generator-maven-plugin` `4.1.3` and the `spring` generator uses an old swagger dependency.

expected:
```
		<dependency>
			<groupId>io.swagger.core.v3</groupId>
			<artifactId>swagger-annotations</artifactId>
			<version>2.0.10</version>
		</dependency>
```

actual:
```
		<dependency>
			<groupId>io.swagger</groupId>
			<artifactId>swagger-annotations</artifactId>
			<version>1.5.23</version>
		</dependency>
```

And the generated code uses `import io.swagger.annotations.*` 
instead of `import io.swagger.v3.oas.annotations.*`

**Comment by wing328** [comment]
@dkirrane yes, we've not yet updated the annotation with io.swagger.v3.oas.annotations

Is that something you can help us with a PR?

**Comment by N4zroth** [comment]
Would you want to still have the possibility to generate "old" code, or do you want to replace it with  the newer version?
Additionally, springfox isn't really maintained anymore, so a transistion to e. g. springdoc would make sense.

**Comment by wing328** [comment]
My take is to replace it with the newer version. If users want the older version, they can use an old version of OpenAPI Generator for that.

**Comment by petervandenbroek** [comment]
Any idea on when this will be fixed?

Currently, generating code from an OAP3 spec leaves you no other option than to expose a Swagger 2 spec.

**Comment by pynivollet** [comment]
Hello, I think I may need this modification for my project, because for now I am not able to use OpenApi3 specification when I generate code from my spring-boot project. Do we have any idea of when it will be done? Thanks

**Comment by cbornet** [comment]
> Currently, generating code from an OAP3 spec leaves you no other option than to expose a Swagger 2 spec.

This is not due to the annotations. You can perfectly generate an OAS3 from old swagger annotations. The fact that a Swagger2 spec is exposed comes from the lib used to generate it (probably springfox v2 ?). You can upgrade to Springfox v3 which supports generating OAS3 specs.
